File: go/difference-between-ones-and-zeros-in-row-and-column.go
func onesMinusZeros(grid [][]int) [][]int {
    n, m := len(grid), len(grid[0])
    ones_row := make([]int, n)
    ones_col := make([]int, m)
    for i := 0; i < n; i++ {
        for j := 0; j < m; j++ {
            is_one := grid[i][j]
            ones_row[i] += is_one
            ones_col[j] += is_one
        }
    }
    for i := 0; i < n; i++ {
        for j := 0; j < m; j++ {
            grid[i][j] = 2 * ones_row[i] + 2 * ones_col[j] - n - m
        }
    }
    return grid
}

File: python/difference-between-ones-and-zeros-in-row-and-column.py
class Solution:
    def onesMinusZeros(self, grid: List[List[int]]) -> List[List[int]]:
        n, m = len(grid), len(grid[0])
        ones_row = [0] * n
        ones_col = [0] * m
        for i in range(n):
            for j in range(m):
                is_one = grid[i][j]
                ones_row[i] += is_one
                ones_col[j] += is_one
        for i in range(n):
            for j in range(m):
                grid[i][j] = 2 * ones_row[i] + 2 * ones_col[j] - n - m
        return grid
